python_modules/dagster/dagster/_core/ext/subprocess.py

```diff
@@ -88,3 +84,6 @@ def _setup_io(
                 message_reader.read_messages(external_context)
             )
             yield io_params_as_env_vars(context_injector_params, message_reader_params)
+
+
+ExtSubprocess = ResourceParam[_ExtSubprocess]
```

Comment:
@benpankow wanted to flag this. I think this is a way to avoid forcing people from doing `ResourceParam` everything. We could probably support this directly with a marker interface or something, but this was easy for now.

Reply:
This seems reasonable to me and is a neat pattern - I wonder if this sort of approach could help in cases where folks are using `ResourceParam` right now. Maybe if they re-use the same POPO in various places...?

Reply:
Yeah I guess I couldn't write this in docs without feeling embarassed about it. It is very counterintuitive at first glance.